src/components/results/TableByQuestion.js

```diff
@@ -0,0 +1,189 @@
+import { useState } from 'react';
+
+import { CancelOutlined, CheckCircleOutlined } from '@mui/icons-material';
+import {
+  Paper,
+  Stack,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TableSortLabel,
+  Typography,
+} from '@mui/material';
+import Box from '@mui/material/Box';
+import { visuallyHidden } from '@mui/utils';
+
+import { QUESTION_TYPES } from '../../config/constants';
+import { computeCorrectness } from '../context/utilities';
+
+const TableByQuestion = ({ question, userList, responses }) => {
+  // TODO Add translation plugin
+
+  /**
+   * Object that represent an enum with the possible value for ordering property
+   *
+   * @type {Readonly<{ASC: string, DESC: string}>}
+   */
+  const Order = Object.freeze({
+    ASC: 'asc',
+    DESC: 'desc',
+  });
+
+  const [order, setOrder] = useState(Order.ASC);
+
+  /**
+   * Helper function to extract the response from a particular user
+   *
+   * @param {string} userId The name of the user
+   * @returns A user response (May be undefined if user didn't respond to the current question)
+   */
+  const getResponseForUserId = (userId) => {
+    return responses.find(({ memberId }) => memberId === userId);
+  };
+
+  /**
+   * Helper function to extract the data of one user for a specific question.
+   *
+   * @param {string} userId The name of the user
+   * @returns {string} Response for given user.
+   */
+  const getResponseDataForUserId = (userId) => {
+    const { text, value, choices } = getResponseForUserId(userId)?.data ?? {
+      text: '',
+    };
```

Comment:
Your util functions are useful but not really efficient. You use three of them in the render, which means you run 3 "loops" per render.
The most optimised way in my opinion would be:
- from all the appData, use `groupBy` with `memberId` (groupBy exists on immutable's List)
- loop over the `groupBy`ed value, it will be an array, so you will need to get the latest one. -> this change will also help you when displaying multiple answers from a same member.
- Update your util functions accordingly so they take one appData as argument (instead of the user id).

Reply:
Yes that is a very good point, thank you for pointing that out! 

I updated my function to use a `Map` that group all answers to a question by their `memberId`.

As for now the user can only provide a single answer for a question, I simply took the `first()` of the list, I will adapt that later, when we add support for giving multiple answers.